boa/src/syntax/ast/node.rs

```diff
@@ -0,0 +1,300 @@
+                    join_expr(f, &e.1)?;
+                }
+                f.write_str("default: \n")?;
+                Display::fmt(def, f)?;
+                f.write_str("}")
+            }
+            Node::ObjectDecl(ref map) => {
+                f.write_str("{")?;
+                for (key, value) in map.iter() {
+                    f.write_fmt(format_args!("{}: {},", key, value))?;
+                }
+                f.write_str("}")
+            }
+            Node::ArrayDecl(ref arr) => {
+                f.write_str("[")?;
+                join_expr(f, arr)?;
+                f.write_str("]")
+            }
+            Node::FunctionDecl(ref name, ref _args, ref expr) => {
+                write!(f, "function ")?;
+                if let Some(func_name) = name {
+                    f.write_fmt(format_args!("{}", func_name))?;
+                }
+                write!(f, "{{")?;
+                // join_expr(f, args)?;
+                write!(f, "}} {}", expr)
+            }
+            Node::ArrowFunctionDecl(ref _args, ref expr) => {
+                write!(f, "(")?;
+                // join_expr(f, args.ini)?;
+                write!(f, ") => {}", expr)
+            }
+            Node::BinOp(ref op, ref a, ref b) => write!(f, "{} {} {}", a, op, b),
+            Node::UnaryOp(ref op, ref a) => write!(f, "{}{}", op, a),
+            Node::Return(Some(ref ex)) => write!(f, "return {}", ex),
+            Node::Return(None) => write!(f, "return"),
+            Node::Throw(ref ex) => write!(f, "throw {}", ex),
+            Node::Assign(ref ref_e, ref val) => write!(f, "{} = {}", ref_e, val),
+            Node::VarDecl(ref vars) | Node::LetDecl(ref vars) => {
+                if let Node::VarDecl(_) = *self {
+                    f.write_str("var ")?;
+                } else {
+                    f.write_str("let ")?;
+                }
+                for (key, val) in vars.iter() {
+                    match val {
+                        Some(x) => f.write_fmt(format_args!("{} = {}", key, x))?,
+                        None => f.write_fmt(format_args!("{}", key))?,
+                    }
+                }
+                Ok(())
+            }
+            Node::ConstDecl(ref vars) => {
+                f.write_str("const ")?;
+                for (key, val) in vars.iter() {
+                    f.write_fmt(format_args!("{} = {}", key, val))?
+                }
+                Ok(())
+            }
+            Node::TypeOf(ref e) => write!(f, "typeof {}", e),
+        }
+    }
+}
+
+/// `join_expr` - Utility to join multiple Expressions into a single string
+fn join_expr(f: &mut Formatter, expr: &[Node]) -> Result {
+    let mut first = true;
+    for e in expr.iter() {
+        if !first {
+            f.write_str(", ")?;
+        }
+        first = false;
+        Display::fmt(e, f)?;
+    }
+    Ok(())
+}
+
+// https://tc39.es/ecma262/#prod-FormalParameter
```

Comment:
I think this should be added as a doc comment, and not only as a comment. Giving some explanations on the implementation and adding the link for further clarifications on what this is.

Reply:
```suggestion
/// "Formal parameter" is a fancy way of saying "function parameter".
///
/// In the declaration of a function, the parameters must be identifiers,
/// not any value like numbers, strings, or objects.
///```javascript
///function foo(formalParametar1, formalParametar2) {
///}
///```
/// https://tc39.es/ecma262/#prod-FormalParameter
```

Reply:
This is great!

The link should be between `<` and `>`, and maybe have some explanatory text, before:
`for more information, please check <https://tc39.es/ecma262/#prod-FormalParameter>`

I think this is a great opportunity to greatly improve the documentation of the parser :)

Reply:
Agreed.